non-critical-infra/modules/vaultwarden.nix

```
@@ -92,6 +92,52 @@
 
 
   };
+
+  # Local mailer
+
+  sops.secrets.opendkim-private-key = {
+    sopsFile = ../secrets/opendkim-private-key.caliban;
+    format = "binary";
+    owner = config.services.postfix.user;
+  };
+  services.opendkim = {
+    enable = true;
+    domains = "vault.nixos.org";
+    selector = "mail";
+    user = config.services.postfix.user;
+    group = config.services.postfix.group;
+    keyPath = "/run/opendkim-keys";
+  };
+
+  systemd.services.opendkim.serviceConfig = {
+    ExecStartPre = [
+      (
+        "+${pkgs.writeShellScript "opendkim-keys" ''
+          install -o ${config.services.postfix.user} -g ${config.services.postfix.group} -D -m0700 ${config.sops.secrets.opendkim-private-key.path} /run/opendkim-keys/${config.services.opendkim.selector}.private
+        ''}"
+      )
+    ];
+  };
+
+  services.postfix = {
+    enable = true;
+    hostname = "vault.nixos.org";
+    domain = "vault.nixos.org";
```

```suggestion
    hostname = config.networking.fqdn;
    domain = config.networking.fqdn;
```

Let's generalize this into a `postfix.nix` and use the fqdn as the sender domain. Because it is very likely that we'll have other services on the same host wanting to send mail, and then service@fqdn would be the sender address of choice.